src/ui/pages/orders/components/orderDetailsReceivedProducts.page.ts

```diff
@@ -186,4 +186,16 @@ export class OrderDetailsReceivedProductsSection extends SalesPortalPage {
   async isSelectAllCheckboxChecked() {
     return await this.selectAllCheckbox.isChecked();
   }
+
+  async getAllProductNames() {
+    const count = await this.allProductAccordionHeaders.count();
+    const names = [];
+    for (let i = 0; i < count; i++) {
+      const header = this.allProductAccordionHeaders.nth(i);
+      const button = header.locator('button.accordion-button');
+      const name = await button.innerText();
+      names.push(name.trim());
+    }
+    return names;
+  }
```

Comment:
Есть возможно собирать сразу все элеметы без count и проходить отдельно по каждому
И можем использовать Promise.all для оптимизации
```suggestion
  async getAllProductNames() {
    const namesLocator = await this.allProductAccordionHeaders.locator('#products-accordion-section .accordion-header button').all();
  
  const names = await Promise.all(
    locators.map(async (locator) => const text = await locator.innerText().trim())
  );
    return names;
  }
```